evutil.c

```diff
@@ -3030,3 +3030,124 @@ evutil_free_globals_(void)
 	evutil_free_secure_rng_globals_();
 	evutil_free_sock_err_globals();
 }
+
+int 
+evutil_set_tcp_keepalive(evutil_socket_t fd, int on, int timeout)
+{
+    int idle;
+    int intvl;
+    int cnt;
+
+    /* Prevent compiler from complaining unused variables warnings. */
+    (void) idle;
+    (void) intvl;
+    (void) cnt;
```

Comment:
Can you move this into relevant sections and only needable variables, so that we will introduce any issues here

Reply:
Sorry, I don't quite get you, did you mean that eliminate these lines and declare/define these variables inside the macros?

Reply:
At first yes, but then I saw that they are just `#ifdef SO_KEEP...` while I thought they will be like `#ifdef __linux__` (or smth similar), so OK, let's leave as is.